syntax = "proto2";

package asn1_primitive_types;

message Identifier {
  required Class id_class = 2;
  required Encoding encoding = 1;
}

// See X.690 (2015), 8.1.2.2.
enum Class {
  Universal = 0;
  Application = 1;
  ContextSpecific = 2;
  Private = 3;
}

// See X.690 (2015), 8.1.2.5.
enum Encoding {
  Primitive = 0;
}

message BitString {
  required Identifier id = 1;
  required bytes val = 2;
}

message Integer {
  required Identifier id = 1;
  required bytes val = 2;
}